modules.d/62bluetooth/module-setup.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# called by dracut
+check() {
+  require_binaries /usr/libexec/bluetooth/bluetoothd || return 1
+  # Include by default if a Peripheral (0x500) is found of minor class:
+  #  * Keyboard (0x40)
+  #  * Keyboard/pointing (0xC0)
+  grep -qiE 'Class=0x[0-9a-f]{3}5[4c]0' /var/lib/bluetooth/*/*/info 2>/dev/null && return 0
+
+  return 255
+}
+
+depends() {
+  echo systemd dbus
+  return 0
+}
+
+installkernel() {
+  instmods bluetooth btrtl btintel btbcm bnep ath3k btusb rfcomm hidp
+  inst_multiple -o \
+    /usr/lib/firmware/ar3k/AthrBT* \
+    /usr/lib/firmware/ar3k/ramps* \
+    /usr/lib/firmware/ath3k-1.fw \
+    /usr/lib/firmware/BCM2033-MD.hex \
+    /usr/lib/firmware/bfubase.frm \
+    /usr/lib/firmware/BT3CPCC.bin \
+    /usr/lib/firmware/brcm/*.hcd \
+    /usr/lib/firmware/mediatek/mt7622pr2h.bin \
+    /usr/lib/firmware/qca/nvm* \
+    /usr/lib/firmware/qca/crnv* \
+    /usr/lib/firmware/qca/rampatch* \
+    /usr/lib/firmware/qca/crbtfw* \
+    /usr/lib/firmware/rtl_bt/* \
+    /usr/lib/firmware/intel/ibt* \
+    /usr/lib/firmware/ti-connectivity/TIInit_* \
+    /usr/lib/firmware/nokia/bcmfw.bin \
+    /usr/lib/firmware/nokia/ti1273.bin
+}
+
+install() {
+  inst_multiple \
+    /usr/libexec/bluetooth/bluetoothd \
+    /usr/lib/systemd/system/bluetooth.target \
+    /usr/lib/systemd/system/bluetooth.service \
+    /etc/bluetooth/main.conf \
+    /etc/dbus-1/system.d/bluetooth.conf \
+    bluetoothctl
+
+  inst_multiple $(find /var/lib/bluetooth)
+
+  inst_rules 69-btattach-bcm.rules 60-persistent-input.rules
+
+  sed -i -e \
+'/^\[Unit\]/aDefaultDependencies=no\
+Conflicts=shutdown.target\
+Before=shutdown.target\
+After=dbus.service' \
+    "$initdir"/usr/lib/systemd/system/bluetooth.service
```

Comment:
We might end up having to include our own unit here.

Reply:
The [upstream unit](https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/src/bluetooth.service.in) is non-trivial , so it may make more sense to keep modifying it.

(I did not author this part, it came from https://github.com/dracutdevs/dracut/pull/544 ).